internal/cli/server/config.go

```diff
@@ -551,6 +553,13 @@ type CacheConfig struct {
 	FDLimit int `hcl:"fdlimit,optional" toml:"fdlimit,optional"`
 }
 
+type LevelDbConfig struct {
+	LevelDbCompactionTableSize           uint64  `hcl:"compactiontablesize,optional" toml:"compactiontablesize,optional"`
+	LevelDbCompactionTableSizeMultiplier float64 `hcl:"compactiontablesizemultiplier,optional" toml:"compactiontablesizemultiplier,optional"`
+	LevelDbCompactionTotalSize           uint64  `hcl:"compactiontotalsize,optional" toml:"compactiontotalsize,optional"`
+	LevelDbCompactionTotalSizeMultiplier float64 `hcl:"compactiontotalsizemultiplier,optional" toml:"compactiontotalsizemultiplier,optional"`
+}
```

Comment:
I was confused by the duplication of parameters in multiple places, as well. For example, in the existing `CacheConfig` struct above you can see the same cache configs that are defined in [`cmd/utils/flags.go`.](https://github.com/maticnetwork/bor/blob/95953adb6c4082b89fb1b09f609d19040030757f/internal/cli/server/config.go#L516).

The only difference as far as I can tell are the field tags, which do indeed seem specific to configuration and would be out of place in `ethdb/leveldb/leveldb.go`. Thoughts?

Reply:
Agreed it is a bit confusing. Configs defined in `internal/cli/server/config.go` are duplicates of `cmd/utils/flags.go`. This was due to a migration of cli since bor 0.3.0. Since `cmd/utils/flags.go` are not used in bor cli, I don't think we need to `cmd` module. On second thought, these flags in `internal/cli/server/config.go` are required, because we use them to parse cli and toml config files. The only thing I would change here is renaming `LevelDBConfig` to `ExtraDBConfig` like you did elsewhere.